src/navigators/SignUpStackNavigator.tsx

```diff
@@ -47,6 +45,9 @@ export const SignUpStackNavigator = () => (
       component={TermsOfServiceScreen}
     />
     <Stack.Screen name={ROUTES.REFERRAL} component={ReferralScreen} />
-    <Stack.Screen name={ROUTES.ONBOARDING} component={OnboardingScreen} />
+    <Stack.Screen
+      name={ROUTES.ONBOARDING}
+      component={OnboardingScreenForNavigator}
```

Comment:
This is to get around having to pass a prop to the component. It's much easier this way, even if it's a little bit of a workaround.